src/Data/RFC5322.hs

```diff
@@ -179,6 +186,20 @@ data Mailbox =
              AddrSpec
     deriving (Show,Eq)
 
+renderMailboxes :: [Mailbox] -> B.ByteString
+renderMailboxes xs = B.intercalate ", " $ renderMailbox <$> xs
+
+renderMailbox :: Mailbox -> B.ByteString
+renderMailbox (Mailbox n a) =
+    let name =
+            maybe
+                mempty
+                (\x ->
+                      mconcat ["\"", Builder.byteString $ T.encodeUtf8 x, "\" "])
+                n
+        builder = mconcat [name, "<", renderAddressSpec a, ">"]
+    in toStrict $ Builder.toLazyByteString builder
```

Comment:
I don't think `Builder` buys us much here, unless it's used for the whole mailbox list.  If you still want to export `renderMailbox` and `renderMailboxes` with their current types, write an internal function (e.g. `buildMailbox :: Mailbox -> Builder`) and use it from both `renderMailbox` and `renderMailboxes`.